modules/claims_api/app/models/claims_api/form_526.rb

```diff
@@ -38,23 +40,29 @@ class Form526
     end
 
     def initialize(params = {})
-      @attributes = []
-      params.each do |name, value|
-        @attributes << name.to_sym
-        begin
-          send("#{name}=", value)
-        rescue StandardError
+      sanitized_params = sanitize_fields(params)
+      sanitized_params.each do |name, value|
+        send("#{name}=", value)
+      end
+    end
+
+    def sanitize_fields(params)
+      bad_fields = params.keys.to_a - ALL_FIELDS.map & :to_sym
+      if bad_fields.any?
+        bad_fields.each do |name|
           errors.add(name, 'is not a valid attribute')
         end
       end
+      params.delete_if { |k, _v| bad_fields.include?(k) }
+      params
     end
 
     def persisted?
       false
     end
 
     def attributes
-      @attributes.map { |method| { method => send(method) } }.reduce(:merge)
+      ALL_FIELDS.map { |method| { method => send(method) } }.reduce(:merge).delete_if { |_k, v| v.blank? }
```

Comment:
Could we get the same effect of removing keys with empty values by doing [Hash.compact](https://apidock.com/rails/Hash/compact) (or a recursive version of it)?

Reply:
Will consider that for a refactor.